common/common/src/main/java/io/helidon/common/SerializationConfig.java

```diff
@@ -0,0 +1,624 @@
+import java.util.stream.Collectors;
+
+/**
+ * Support for JEP 290 - deserialization filtering.
+ * Configuration options mentioned below will differ in Helidon 3.0.0, the following table lists the options:
+ * <p>
+ * <table class="config">
+ * <caption>Configuration Options</caption>
+ * <tr>
+ *     <th>system property</th>
+ *     <th>2.x default</th>
+ *     <th>3.x default</th>
+ *     <th>description</th>
+ * </tr>
+ * <tr>
+ *     <td>{@link #PROP_WRONG_CONFIG_ACTION}</td>
+ *     <td>{@code warn} - {@link io.helidon.common.SerializationConfig.Action#WARN}</td>
+ *     <td>{@code fail} - {@link io.helidon.common.SerializationConfig.Action#FAIL}</td>
+ *     <td>What to do if an existing global deserialization filter exists without a global blacklist.</td>
+ * </tr>
+ * <tr>
+ *     <td>{@link #PROP_NO_CONFIG_ACTION}</td>
+ *     <td>{@code warn} - {@link io.helidon.common.SerializationConfig.Action#WARN}</td>
+ *     <td>{@code configure} - {@link io.helidon.common.SerializationConfig.Action#CONFIGURE}</td>
+ *     <td>What to do if there is no global deserialization filter.</td>
+ * </tr>
+ * </table>
+ * Last option (not used by default) is to {@code ignore} the problem and do nothing (can be used both with wrong config
+ * and no config above).
+ * <h2>Deserialization filtering in Helidon</h2>
+ * Helidon serialization filter is implemented to support whitelists, automatically blacklisting
+ * all classes.
+ * Helidon restrictions are only enforced on the global filter.
+ * <h3>Custom pattern</h3>
+ * To add patterns to the serial filter, use a system property {@value #PROP_PATTERN}.
+ * This pattern follows the rules as defined by JDK. Helidon will add exclude all as the last
+ * pattern.
+ * <p>
+ * As an alternative, a file {@link #PROPERTY_FILE} can be created on the classpath with the following content, to
+ * configure filter for a specific library. Do not add a global blacklist to these patterns!:
+ * {@code pattern=oracle.sql.converter.*}
+ * <h3>Deserialization tracing</h3>
+ * A tracing filter can be configured using system property {@value #PROP_TRACE} to log information
+ * messages for each deserialization request.
+ * <p>
+ * To discover class patterns needed, set "no config" and "wrong config" actions to warn or ignore, and configure {@code basic}
+ * tracing.
+ * <p>
+ * Options are:
+ * <ul>
+ *     <li>{@code none} to disable tracing (this is the default)</li>
+ *     <li>{@code basic} to enable basic tracing (only traces requests for class deserialization)</li>
+ *     <li>{@code full} to enable full tracing (including sizes, depth etc.)</li>
+ * </ul>
+ */
+public final class SerializationConfig {
+    static final String PROP_WRONG_CONFIG_ACTION = "helidon.serialFilter.failure.action";
+    static final String PROP_NO_CONFIG_ACTION = "helidon.serialFilter.missing.action";
+    static final String PROP_PATTERN = "helidon.serialFilter.pattern";
+    static final String PROP_TRACE = "helidon.serialFilter.trace";
```

Comment:
I wonder if this could be triggered off of log levels instead of inventing a new property. For example:
helidon.serialFilter.trace=basic -> io.helidon.serialFilter.level=INFO 
helidon.serialFilter.trace=full -> io.helidon.serialFilter.level=FINE

Maybe that complicates the implementation too much.

Reply:
Problem with logging is that we may not be using Java Util logging at runtime - it may be log4j, logback etc. And I am not sure we could even reach this information from code

Reply:
the reason it is done in advance is for performance - I do not have tracing code on code path at all, unless you enable it